ProQuest.js

```diff
@@ -118,59 +121,95 @@ function fetchEmbeddedPdf(url, item, callback) {
+			break;
+		}
+	}
+	if (!root) {
+		var elements = doc.getElementsByClassName('resultListContainer');
+		for (var i=0; i<elements.length; i++) {
+			if (elements[i] && elements[i].offsetHeight>0) {
+				root = elements[i];
 				break;
 			}
 		}
-	} else {
-		root = doc.getElementsByClassName('resultListContainer')[0];
-		if (root && !root.offsetHeight) return false; // Not sure if this can actually happen
 	}
-
-	if (!root) return false;
+	if (!root) {
+		Z.debug("No root found");
+		return false;
+	}
 
 	var results = doc.getElementsByClassName('resultItem');
-	//ZU.xpath(root, './/a[contains(@class,"previewTitle") or contains(@class,"resultTitle")]');
-	
+
 	var items = {}, found = false;
 	for(var i=0, n=results.length; i<n; i++) {
-		var title = results[i].getElementsByClassName('resultTitle')[0]
-			|| results[i].getElementsByClassName('previewTitle')[0];
-		if (!title || title.nodeName != 'A') continue;
-		
-		if (checkOnly) return true;
-		found = true;
-		
-		items[title.href] = title.textContent;
+		if (results[i].offsetHeight>0) { //restrict to the active results only
+			var title = results[i].getElementsByClassName('previewTitle')[0]
+				|| results[i].getElementsByClassName('resultTitle')[0];
+			if (!title || title.nodeName != 'A') continue;
+			
+			if (checkOnly) return true;
+			found = true;
+			
+			items[title.href] = title.textContent;
+		}
 	}
 		
 	return found ? items : false;
 }
 
+var map = {
+	"newspaperArticle" : "newspaperArticle",
+	"jnlArticle" : "journalArticle", //also magazineArticles
+	"dissertationPublished" : "thesis",
+	"reportSerial" : "report",
+	"workingPaper" : "report",
+	"book" : "book",
+	"webPage" : "webpage",
+	"blog" : "blogPost",
+	"report" : "report",
+	"audioVideo" : "videoRecording",
+	"default" : "conferencePaper" //also default = manuscript?, referenceWork, encyclopedia
+}
+
 function detectWeb(doc, url) {
 	initLang(doc, url);
-
-	followLink = false;
-
+	
+	var abstractTab = doc.getElementById('tab-AbstractRecord-null'); // Seems like that null is a bug and it might change at some point
+	followLink = !(abstractTab && abstractTab.classList.contains('active'));
+	
 	//Check for multiple first
-	if (url.indexOf('docview') == -1 &&
-		url.indexOf('pagepdf') == -1) {
-		if (getSearchResults(doc, true))
-			return "multiple";
+	if (url.indexOf('docview') == -1 && url.indexOf('pagepdf') == -1) {
+		return getSearchResults(doc, true) ? 'multiple' : false;
 	}
-
+	
+	//if we are on Abstract/Details page,
+	//then we can read the type from the corresponding field
 	var types = getTextValue(doc, ["Source type", "Document type", "Record type"]);
 	var zoteroType = getItemType(types);
-	if(zoteroType) return zoteroType;
-
+	if (zoteroType) return zoteroType;
+
+	//search for comments like
+	//<!-- Open:block:newspaperArticle -->
+	//these are also present on the pdf sites
+	var html = doc.body.innerHTML;
+	var matches = html.match(/Open:block:\S+/);
```

Comment:
I can't follow here. On pdf pages we need to determine somehow its type and the code here works okay.